apps/api/src/app/shared/interceptors/cached.interceptor.ts

```diff
@@ -0,0 +1,52 @@
+import { buildCachedQuery, buildKey, CacheInterceptorTypeEnum } from './shared-cache';
+import { Inject } from '@nestjs/common';
+import { CacheKeyPrefixEnum, CacheService } from '../services/cache';
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export function Cached(storeKeyPrefix: CacheKeyPrefixEnum) {
+  const injectCache = Inject(CacheService);
+
+  return (target: any, key: string, descriptor: any) => {
+    const originalMethod = descriptor.value;
+    const methodName = key;
+    injectCache(target, 'cacheService');
+
+    descriptor.value = async function (...args: any[]) {
+      const skip = args[2]?.skipCache;
+      if (!this.cacheService?.cacheEnabled() || skip) return await originalMethod.apply(this, args);
+
+      const query = buildCachedQuery(args);
+
+      const cacheKey = buildKey(storeKeyPrefix, query, CacheInterceptorTypeEnum.CACHED);
+
+      if (!cacheKey) {
+        return await originalMethod.apply(this, args);
+      }
```

Comment:
to me, this should never happen if everything was implemented properly in the `buildKey`, like what's the point of using the `@Cached` annotation if the result won't be cached at all

Reply:
Tend to agree, I created it as a guard when we did the DAL implementation. Do you think we should remove it?